lib/graphql/schema/input_object.rb

```diff
@@ -53,6 +53,15 @@ def to_hash
         to_h
       end
 
+      def deconstruct_keys(keys = nil)
+        if keys.nil?
+          @ruby_style_hash
+        else
+          keys.select { @ruby_style_hash.key?(_1) }
+              .to_h { [_1, @ruby_style_hash[_1]] }
```

Comment:
I'm interested in improving the performance of this before we merge it. I think we can reduce the allocations to _just_ the newly-returned hash. How about this? 

```suggestion
          new_h = {}
          keys.each { |k| @ruby_style_hash.key?(k) && new_h[k] = @ruby_style_hash[k] }
          new_h 
```